_docs-sources/pipelines/tutorial/index.md

```diff
@@ -1,19 +1,358 @@
-Just like how you need to monitor your cat's health to make sure they're happy and healthy, you need to monitor your pipeline to make sure it's working properly. Keep an eye on the pipeline's logs and metrics to make sure everything is running smoothly. If you notice any issues, make improvements to your pipeline script to fix them.
+## Step 1: Repo Setup
+
+The code for this tutorial can be found in the [Gruntwork Service Catalog](https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/master/examples/for-learning-and-testing/gruntwork-pipelines/README.md). Start by cloning the repo:
+
+```shell
+git clone https://github.com/gruntwork-io/terraform-aws-service-catalog.git
+```
+
+We will be following the example found at `terraform-aws-service-catalog/examples/for-learning-and-testing/gruntwork-pipelines`
+
+```shell
+cd terraform-aws-service-catalog/examples/for-learning-and-testing/gruntwork-pipelines
+```
+
+## Step 2: Create the required ECR repositories
+
+Change directories to deploy the terraform for ECR
+
+```shell
+cd ecr-repositories
+```
+
+Set the `AWS_REGION` environment variable to your desired AWS region:
+
+```shell
+export AWS_REGION=<your-desired-region>
+```
+
+Authenticate with your AWS account and deploy the terraform code provided to create the three
+ECR repositories.
+
+Initialize terraform to download required dependencies:
+```shell
+terraform init
+```
+
+Run plan and ensure the output matches your expectations:
+```shell
+terraform plan
+```
+
+Deploy the code using apply
+```shell
+terraform apply
+```
+
+## Step 3: Build and Push the Docker Images
+
+The four standard Gruntwork Pipelines capabilities are instrumented by two separate docker files
+
+1. [ecs-deploy-runner](./docker/deploy-runner/Dockerfile) - Terraform plan, apply and AMI building
+2. [kaniko](./docker/kaniko/Dockerfile) - Docker image building. [Kaniko](https://github.com/GoogleContainerTools/kaniko) is a tool that supports building Docker images inside of a container
+
+These Dockerfiles live in the ecs-deploy-runner module within [the terraform-aws-ci repository](https://github.com/gruntwork-io/terraform-aws-ci). In this example, we'll be cloning the terraform-aws-ci and running Docker build against the Dockerfiles defined there.
+
+We're now going to build these two Docker images and push them to the ECR repositories we just created.
+
+### Step 3a: Export Environment Variables
+
+If you do not already have a GitHub Personal Access Token (PAT) available, you can follow this [guide to Create a new GitHub Personal Access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token)
+
+For the purposes of this example, your token will need the `repo` scope, so that Gruntwork Pipelines is able to fetch modules and code from private Gruntwork repositories. Note that in production, the best practice would be to create a separate GitHub machine user account,
```

Comment:
```suggestion
For the purposes of this example, your token will need the `repo` scope, so that Gruntwork Pipelines is able to fetch modules and code from private Gruntwork repositories. Note that in production, the best practice is to create a separate GitHub machine user account,
```